components/form/TimesheetForm.tsx

```diff
@@ -168,37 +171,55 @@ function TimesheetForm({
                     <FormControl isInvalid={invalidAllocation}>
                         <FormLabel>Allocation</FormLabel>
                         <Input
-                            value={timesheetFields.allocation || ""}
+                            type={"number"}
+                            value={
+                                timesheetFields.allocation === undefined
+                                    ? ""
+                                    : timesheetFields.allocation
+                            }
                             placeholder="0"
                             onChange={(event) =>
                                 setTimesheetFields({
                                     ...timesheetFields,
-                                    allocation: Number(event.target.value),
+                                    allocation: event.target.value
+                                        ? Number(event.target.value)
+                                        : undefined,
                                 })
                             }
                             data-testid="form-field-allocation"
                         />
                         <FormErrorMessage>
-                            Allocation needs to be between 1 and 100 %.
+                            Allocation needs to be between 0 and 100 %.
+                        </FormErrorMessage>
+                    </FormControl>
+                    <FormControl isInvalid={!rateIsValid}>
+                        <FormLabel>Rate</FormLabel>
+                        <Input
+                            type={"number"}
+                            value={
+                                timesheetFields.rate === undefined
+                                    ? ""
+                                    : timesheetFields.rate
+                            }
+                            placeholder="Rate (€/h)"
+                            onChange={(event) => {
+                                setTimesheetFields({
+                                    ...timesheetFields,
+                                    rate: event.target.value
+                                        ? Number(event.target.value)
+                                        : undefined,
+                                })
+                            }}
+                            data-testid="form-field-rate"
+                        />
+                        <FormErrorMessage>
+                            Rate must be a positive number.
```

Comment:
Nit: Since zero is allowed, `Rate must be a non-negative number.`

Reply:
Changed this 👍